### PR TITLE
READY: Fix bugs with loading AttestationCommunity

### LINKED ISSUE #117: AttestationCommunity database error
There is an error when loading keys for the AttestationCommunity:
```
04-26 09:57:49.876 4441-4461/? I/IPV8Service: DEBUG:Launching AttestationCommunity with prefix 0002b42c93d167a0fc4a0843f917d4bf1e9ebb340ec4.
04-26 09:57:49.877 4441-4461/? I/IPV8Service: DEBUG:loading database [sqlite/attestations.db]
04-26 09:57:49.877 4441-4461/? I/IPV8Service: DEBUG:open database [sqlite/attestations.db]
04-26 09:57:49.907 4441-4461/? I/IPV8Service: DEBUG:PRAGMA page_size = 8192 (no change) [sqlite/attestations.db]
04-26 09:57:49.907 4441-4461/? I/IPV8Service: DEBUG:PRAGMA journal_mode = WAL (no change) [sqlite/attestations.db]
04-26 09:57:49.907 4441-4461/? I/IPV8Service: DEBUG:PRAGMA synchronous = NORMAL (previously: 2) [sqlite/attestations.db]
04-26 09:57:49.932 4441-4461/? I/IPV8Service: Unhandled Error
04-26 09:57:49.932 4441-4461/? I/IPV8Service: Traceback (most recent call last):
04-26 09:57:49.933 4441-4461/? I/IPV8Service:   File "/data/user/0/org.internetofmoney.android/files/lib/python2.7/site-packages/twisted/internet/base.py", line 417, in fireEvent
04-26 09:57:49.933 4441-4461/? I/IPV8Service:     DeferredList(beforeResults).addCallback(self._continueFiring)
04-26 09:57:49.933 4441-4461/? I/IPV8Service:   File "/data/user/0/org.internetofmoney.android/files/lib/python2.7/site-packages/twisted/internet/defer.py", line 317, in addCallback
04-26 09:57:49.933 4441-4461/? I/IPV8Service:     callbackKeywords=kw)
04-26 09:57:49.933 4441-4461/? I/IPV8Service:   File "/data/user/0/org.internetofmoney.android/files/lib/python2.7/site-packages/twisted/internet/defer.py", line 306, in addCallbacks
04-26 09:57:49.933 4441-4461/? I/IPV8Service:     self._runCallbacks()
04-26 09:57:49.933 4441-4461/? I/IPV8Service:   File "/data/user/0/org.internetofmoney.android/files/lib/python2.7/site-packages/twisted/internet/defer.py", line 588, in _runCallbacks
04-26 09:57:49.933 4441-4461/? I/IPV8Service:     current.result = callback(current.result, *args, **kw)
04-26 09:57:49.933 4441-4461/? I/IPV8Service: --- <exception caught here> ---
04-26 09:57:49.933 4441-4461/? I/IPV8Service:   File "/data/user/0/org.internetofmoney.android/files/lib/python2.7/site-packages/twisted/internet/base.py", line 430, in _continueFiring
04-26 09:57:49.933 4441-4461/? I/IPV8Service:     callable(*args, **kwargs)
04-26 09:57:49.933 4441-4461/? I/IPV8Service:   File "/data/user/0/org.internetofmoney.android/files/lib/python2.7/site-packages/twisted/plugins/ipv8_plugin.py", line 48, in start_ipv8
04-26 09:57:49.933 4441-4461/? I/IPV8Service:     self.ipv8 = IPv8(config)
04-26 09:57:49.933 4441-4461/? I/IPV8Service:   File "/data/user/0/org.internetofmoney.android/files/lib/python2.7/site-packages/ipv8_service.py", line 89, in __init__
04-26 09:57:49.933 4441-4461/? I/IPV8Service:     overlay_instance = overlay_class(my_peer, self.endpoint, self.network, **overlay['initialize'])
04-26 09:57:49.933 4441-4461/? I/IPV8Service:   File "/data/user/0/org.internetofmoney.android/files/lib/python2.7/site-packages/ipv8/attestation/wallet/community.py", line 56, in __init__
04-26 09:57:49.933 4441-4461/? I/IPV8Service:     self.attestation_keys[hash] = BonehPrivateKey.unserialize(key)
04-26 09:57:49.933 4441-4461/? I/IPV8Service: exceptions.TypeError: writable buffers are not hashable
```

### PR DESCRIPTION
**Pull request by qstokkink**
Fixes #117 

 - Fixes loading keys from database
 - Fixes incorrect number of arguments for default completion callback
 - Adds Windows-safe temporary directory creation method

### CODE REVIEW

File: ipv8/attestation/wallet/community.py [hunk 1]
```diff
@@ -48,12 +48,12 @@ def __init__(self, *args, **kwargs):
         super(AttestationCommunity, self).__init__(*args, **kwargs)
 
         self.database = AttestationsDB(working_directory, db_name)
-        self.attestation_request_callbacks = [lambda x, y: None, lambda x, y, z: None]
+        self.attestation_request_callbacks = [lambda x, y: None, lambda x, y, z, w=None: None]
```
Comment (devos50) [inline ipv8/attestation/wallet/community.py]: Two comments: first, I think the naming of the variables should be more clear and precise. Second, I see two callbacks here but I have no idea what they are doing exactly. Could you use a named tuple or even a dictionary instead to store these callbacks by a human-readable key?

Comment (devos50) [inline ipv8/attestation/wallet/community.py]: Two comments: first, I think the naming of the variables should be more clear and precise. Second, I see two callbacks here but I have no idea what they are doing exactly. Could you use a named tuple or even a dictionary instead to store these callbacks by a human-readable key?

File: ipv8/test/base.py [hunk 2]
```diff
@@ -49,10 +51,14 @@ def setUp(self):
...
-        for node in self.nodes:
-            node.unload()
-        internet.clear()
+        try:
+            super(TestBase, self).tearDown()
+            for node in self.nodes:
+                node.unload()
+            internet.clear()
+        finally:
+            shutil.rmtree("temp", ignore_errors=True)
```
Comment (devos50) [inline ipv8/test/base.py]: You are not removing the directory created in the method `temporary_directory`.

File: ipv8/attestation/wallet/community.py (hunk 1 above)
Comment (qstokkink) [reply ipv8/attestation/wallet/community.py]: Good enhancement, but as this is a general improvement and not required for this fix I have moved this to #121 .

File: ipv8/test/base.py (hunk 2 above)
Comment (qstokkink) [reply ipv8/test/base.py]: `temporary_directory` creates a temporary directory in the `temp` folder (see line 139), this entire folder is `rmtree`'d.

**Pull request merged**